libraries/AP_Scripting/generator/src/main.c

```diff
@@ -377,6 +383,42 @@ void sanatize_name(char **dest, char *src) {
     *position = '_';
     position = strchr(position, ':');
   }
+
+  position = strchr(*dest, '.');
+  while (position) {
+    *position = '_';
+    position = strchr(position, '.');
+  }
```

Comment:
Let's break this out as a core helper of "sanatize_character()" or something along those lines (I'm bad at naming) that simply takes a character to replace everywhere and then call that. This is a lot of duplication in here now, and it's more likely to have a bug in the future. (Also once it's a helper the whole thing can be expressed as a for each char in a string of chars to sanatize)

Reply:
fixed